packages/discovery-provider/src/tasks/entity_manager/entities/track.py

```diff
@@ -680,15 +680,16 @@ def download_track(params: ManageEntityParameters):
 
     session = params.session
 
+    has_metadata = isinstance(params.metadata, dict)
     record = TrackDownload(
         txhash=params.txhash,
         blocknumber=params.block_number,
         parent_track_id=parent_track_id,
         track_id=track_id,
         user_id=params.user_id,
-        city=params.metadata.get("city"),
-        region=params.metadata.get("region"),
-        country=params.metadata.get("country"),
+        city=params.metadata.get("city") if has_metadata else None,
```

Comment:
could also default to None

Reply:
isn't that automatic? .get("") defaults to None if no key. but in this case, i'm checking if it's a dict at all

Reply:
oh, right, nice